Steganography-App-CMSC-495/steganography_backend
# Steg-5 - Image Encryption

**Added**
- Core lib image encryption using bit even-odd encoding (evenOddEncryption(data, msg)
- Core lib difference image generation for debugging/viewing the difference between the original image and the encrypted one

**Fixed**
- Problem where images were being transposed due to improper nested loops

**QA**
Run as usual (python src/util_driver.py)
This will generate (or overwrite) the encrypted_image.png and diff.png with the encrypted image and the difference image (represents the color change between the two images, with a scale to be able to see color change as it is at most 1 off on each component which wouldn't be visible, so it is scaled up to 150 by default)

**Notes**
The functions in util_driver.py are good examples of how to utilize the core library functions and do not represent how the program must behave, it is simply a driver to test the core library.
